cluster/memberlist.go

```diff
@@ -11,7 +11,7 @@ import (
 //TODO: this needs to be implemented,we could send a `Request` to the membership actor, but this seems flaky.
 //a threadsafe map would be better
 func getMembers(kind string) []string {
-	res, err := memberlistPID.RequestFuture(&MemberByKindRequest{kind: kind, onlyAlive: true}, 5*time.Second).Result()
+	res, err := memberlistPID.RequestFuture(&MemberByKindRequest{kind: kind, onlyAlive: false}, 5*time.Second).Result()
```

Comment:
I'm not sure about this change.
This would mean that nodes that Consul considers to be unhealthy are included in the result.
And when you call `cluster.Get` with a given ID that could belong to an unhealthy node, it would try to communicate with this unhealthy node.

(I'm not sure about (my own?) comment above the code either "reachable and unreachable" )

Reply:
I'm thinking if nodes appear as critical when they are not, then this might be an effect of this:

```
func New() (*ConsulProvider, error) {
	client, err := api.NewClient(&api.Config{})
	if err != nil {
		return nil, err
	}
	p := &ConsulProvider{
		client:             client,
		ttl:                3 * time.Second,   
		refreshTTL:         1 * time.Second,
		deregisterCritical: 10 * time.Second,
```

Maybe the ttl and refreshTTL are too narrow and the nodes appear as unhealthy when they are not.
I will investigate

Reply:
Thanks for looking into this @rogeralsing The situation I am running into in my tests is caused by a race condition between a _pidCachePartitionActor_ calling `getMember` to resolve the partition actor's address and _partitionActor_ processing a  `memberJoined` event. In the case of flappy consul health, the `getMember` returns different results between the two and the ownership fails to transfer correctly to the new member.